packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -1,32 +1,222 @@
 import { CfnGatewayRoute } from './appmesh.generated';
+import { HttpRouteMatchMethod } from './http-route-match-method';
+import { HttpRoutePathMatch } from './http-route-path-match';
+import { HeaderMatch } from './header-match';
+import { areMatchPropertiesUndefined, validateGprcMatch, validateMetadata } from './private/utils';
+import { QueryParameterMatch } from './query-parameter-match';
 import { Protocol } from './shared-interfaces';
 import { IVirtualService } from './virtual-service';
 
 // keep this import separate from other imports to reduce chance for merge conflicts with v2-main
 // eslint-disable-next-line no-duplicate-imports, import/order
 import { Construct } from '@aws-cdk/core';
 
+
 /**
- * The criterion for determining a request match for this GatewayRoute
+ * Configuration for gRPC gateway route match with the host name.
+ */
+export interface GatewayRouteHostnameMatchConfig {
+  /**
+   * GatewayRoute CFN configuration for host name match.
+   */
+  readonly hostnameMatch: CfnGatewayRoute.GatewayRouteHostnameMatchProperty;
+}
+
+/**
+ * Used to generate host name matching methods.
+ */
+export abstract class GatewayRouteHostname {
+  /**
+   * The value of the host name must match the specified value exactly.
+   *
+   * @param name The exact host name to match on
+   */
+  public static matchingExactly(name: string): GatewayRouteHostname {
```

Comment:
I would prefer naming the class without including `match` and name the method name with `matching` since that is how [`SubjectAlternativeNames`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-appmesh/lib/tls-validation.ts#L174-L185) was previous modeled.

I will review all class that has `matching...()` method and ensure naming conventions are consistent.

Reply:
I would stick to the `*Match` suffix in the classes, and remove the `matching` from the static factory method names.

`SubjectAlternativeNames` is not part of a "match" concept, so I think it's fine that there's a difference in the names.

Reply:
Well noted.